From aab5bd302ff9de69b5e8585ac4051df42049f957 Mon Sep 17 00:00:00 2001
From: Matthew Vernon <mvernon@wikimedia.org>
Date: Mon, 21 Aug 2023 16:10:11 +0100
Subject: [PATCH] Use the old /updates security map for buster (Closes:
 #1050179)

The suite-map and suite-rmap for debian-security are necessary for the
pre-bullseye layout of the security.debian.org archive.

Since bullseye (i.e. after buster), the archive layout has changed,
and these mappings are no longer necessary (indeed, they cause dgit
clone to fail to work with bullseye and later security suites).

Buster is the oldest suite still available on security.debian.org, so
this is the only suite we still need the mapping for.

Signed-off-by: Matthew Vernon <mvernon@wikimedia.org>
---
 dgit | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dgit b/dgit
index 30d76299..1b02f8de 100755
--- a/dgit
+++ b/dgit
@@ -794,8 +794,8 @@ our %defcfg = ('dgit.default.distro' => 'debian',
 	       'dgit-distro.debian.mirror' => 'http://ftp.debian.org/debian/',
  'dgit-distro.debian-security.archive-query' => 'aptget:',
  'dgit-distro.debian-security.mirror' => 'http://security.debian.org/debian-security/',
- 'dgit-distro.debian-security.aptget-suite-map' => 's#-security$#/updates#',
- 'dgit-distro.debian-security.aptget-suite-rmap' => 's#$#-security#',
+ 'dgit-distro.debian-security.aptget-suite-map' => 's#buster-security$#buster/updates#',
+ 'dgit-distro.debian-security.aptget-suite-rmap' => 's#buster$#buster-security#',
  'dgit-distro.debian-security.nominal-distro' => 'debian',
  'dgit-distro.debian.backports-quirk' => '(squeeze)-backports*',
  'dgit-distro.debian-backports.mirror' => 'http://backports.debian.org/debian-backports/',
-- 
2.39.2

